core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -194,35 +212,37 @@ class TransactionCoordinator(brokerId: Int,
                                        pid: Long,
                                        epoch: Short,
                                        partitions: collection.Set[TopicPartition],
-                                       responseCallback: TxnMetadataUpdateCallback): Unit = {
+                                       responseCallback: AddPartitionsCallback): Unit = {
     val errors = validateTransactionalId(transactionalId)
     if (errors != Errors.NONE)
       responseCallback(errors)
     else {
       // try to update the transaction metadata and append the updated metadata to txn log;
       // if there is no such metadata treat it as invalid pid mapping error.
-      val (error, newMetadata) = txnManager.getTransactionState(transactionalId) match {
+      val (error: Errors, coordinatorEpoch: Int, txnMetadata: TransactionMetadata) = txnManager.getTransactionState(transactionalId) match {
```

Comment:
EDIT: @ijuma I realized that in many cases I would return a `null`, and it will cause the annotated type to be `Any` at compilation time. So I think it is actually better to enforce type annotations in this case.